avoidance_car.ino

```diff
@@ -0,0 +1,83 @@
+#include <Smartcar.h>
+
+const int fSpeed   = 70;  
+const int bSpeed   = -70; 
+const int lDegrees = -75; 
+const int rDegrees = 75;  
+const int TRIGGER_PIN = 6;
+const int ECHO_PIN = 7;
+const unsigned int MAX_DISTANCE =100;
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+}
+
+void handleInput(){
+    if(Serial.available()){
+        char input = Serial.read(); 
+                                    
+        switch (input)
+        {
+        case 'l': 
+            car.setSpeed(fSpeed);
+            car.setAngle(lDegrees);
+            break;
+        case 'r': 
+            car.setSpeed(fSpeed);
+            car.setAngle(90);
+            break;
+        case 'f': 
+            car.setSpeed(fSpeed);
+            car.setAngle(0);
+            break;
+        case 'b': 
+            car.setSpeed(bSpeed);
+            car.setAngle(0);
+            break;
+        default: 
+            car.setSpeed(0);
+            car.setAngle(0);
+        }
+    }
+}
+
+void loop()
+{
+   
+    Serial.println(front.getDistance());
+    delay(100);
+    
+    if(front.getDistance() <= 100 && front.getDistance()>0){

```

Comment:
What happens if the first call to `getDistance` returns `0` and the next one `150`? The car stops while it should not.

Reply:
Hi, what do you mean with is not built by CI?
I just deleted the previous pull request (#9) and created a new one with the arduino file inside  ( pull request #10).

When distance is 0 then there is no obstacle detected and the car only stops if getDistance is bigger than 0 and less or equal to 100.

Reply:
> Hi, what do you mean with is not built by CI?

It is not automatically compiled every time you push like the `smartcar.ino` does for example.

> When distance is 0 then there is no obstacle detected and the car only stops if getDistance is bigger than 0 and less or equal to 100.

Read the problem statement carefully:

> What happens if the first call to getDistance returns 0 and the next one 150?

Do you agree that the car will stop in that case?